components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -394,6 +394,13 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 return;
             }
 
+            // check if the user has access to the project
+            if (
```

Comment:
Do you think we also need to check this inside WorkspaceService.startWorkspace to guard the access [here](https://github.com/gitpod-io/gitpod/blob/d2b220f11cee9ac07688ee802fafe6c8dcc80c75/components/server/src/workspace/workspace-factory.ts#L55-L59)? :thinking:

Reply:
I'll check, but I wonder how a prebuild is relevant when starting an existing workspace.

Reply:
Sorry, I meant WorkspaceService.**create**Workspace :roll_eyes: